library_information_validator.py

```diff
@@ -0,0 +1,495 @@
+                split_cindex = re.split('[-_]',cindex_value)
+                if(len(split_cindex) > 1):
+                    if(len(split_cindex[0]) != len(split_cindex[1])):
+                        logger.warning("Length of the two components in dual index {} and {} is different.".format(split_cindex[0], split_cindex[1]))
+                        warning_component_length += 1
+
+                # merges dual indexes to one sequence
+                index_only = re.sub('[-]','', cindex_value)
+                cindex_list.append(index_only)
+
+        # identifies missing index specification for individual samples
+        # and detects wether NGI standard indexes or custom indexes are chosen for a given pool
+        error_missing_index = 0
+        warning_mixed_indexes = 0
+        sindex_chosen = True
+        if(len(cindex_absent) == len(pool_rows)):
+            sel_index = sindex_list
+            if(len(sindex_list) != len(pool_rows)):
+                for absent_index in sindex_absent:
+                    logger.error("missing index in row {}".format(re.sub("\D","",absent_index)))
+                    quit()
+        elif(len(sindex_absent) == len(pool_rows)):
+            sel_index = cindex_list
+            sindex_chosen = False
+            if(len(cindex_list) != len(pool_rows)):
+                for absent_index in cindex_absent:
+                    logger.error("missing index in row {}".format(re.sub("\D","",absent_index)))
+                    quit()
+        else:
+            sindex_chosen = False
+
+            # warning if a pool consists out of a mix of NGI standard indexes and custom indexes
+            sel_index = sindex_list + cindex_list
+            if(len(sel_index) != len(pool_rows)):
+                rowID_cindex_absent = [re.sub("\D","",x) for x in cindex_absent]
+                rowID_sindex_absent = [re.sub("\D","",x) for x in sindex_absent]
+
+                missing_indexes = list(set(rowID_cindex_absent).intersection(rowID_sindex_absent))
+                for mIndex in missing_indexes:
+                    logger.error('missing index in row {}'.format(mIndex))
+
+            logger.warning('mix between custom and standard indexes in pool {}.'\
+            .format(pool))
+            warning_mixed_indexes += 1
+
+        # returns warnings for check summary
+        return(sel_index, sindex_chosen, warning_mixed_indexes, warning_component_length)
+
+    def validate_index(self, index_seq, pool_name, sindex):
+        '''
+        does all the fancy index checks
+        '''
+
+        # allows for entry "noIndex" if only one sample is defined in the pool
+        c = Counter(index_seq)
+        if(c['noIndex'] > 0 and len(index_seq) != 1):
+            logger.error('Pool {} contains undefined index(es) (\"noIndex\")'\
+            .format(pool_name))
+            quit()
+        elif(c['noIndex'] == 1 and len(index_seq) == 1):
+            logger.info('Pool {} containing one sample is not indexed.'\
```

Comment:
Ack, I hate PEP8 short code lines 😆 I always ignore this style guide and make my lines longer if it makes it easier to read (as it would here I think).